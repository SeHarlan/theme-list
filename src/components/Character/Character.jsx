import React from 'react';
import PropTypes from 'prop-types';

const Character = ({
  status,
  species,
  name,
  image
}) => (
  <>
    <h2>{name}</h2>
    <img src={image} alt={name}/>
    <p>Species: {species}</p>
    <p>Status: {status}</p>
  </>
);

Character.propTypes = {
  status: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default Character;
