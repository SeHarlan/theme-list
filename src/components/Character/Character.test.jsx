import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders Character', () => {
    const wrapper = shallow(<Character image="test" name="test" species="test" status="test"/>);
    expect(wrapper).toMatchSnapshot();
  });
});
