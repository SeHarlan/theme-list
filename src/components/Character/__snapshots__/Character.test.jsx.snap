// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<Fragment>
  <h2>
    test
  </h2>
  <img
    alt="test"
    src="test"
  />
  <p>
    Species: 
    test
  </p>
  <p>
    Status: 
    test
  </p>
</Fragment>
`;
